### PR TITLE
Add missing extends in mob interfaces

### PR DESCRIPTION
**Pull request by ann4belle**
I think I caught most of the interface extensions that were missing. Improved my earlier PR #3794, resubmitted due to scope change.

### CODE REVIEW

File: Spigot-API-Patches/0217-Add-missing-extends.patch [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+diff --git a/src/main/java/org/bukkit/entity/Vex.java b/src/main/java/org/bukkit/entity/Vex.java
+index c34a3ea7b4d16817b4bee25d5c69787e22ec44d8..39236fa222de62ec1193ec956809b56d155e8f57 100644
+--- a/src/main/java/org/bukkit/entity/Vex.java
++++ b/src/main/java/org/bukkit/entity/Vex.java
+@@ -5,7 +5,7 @@ import org.jetbrains.annotations.Nullable;
+ /**
+  * Represents a Vex.
+  */
+-public interface Vex extends Monster {
++public interface Vex extends Monster, Flying { // Paper - Vexes fly
```
Comment (aikar) [inline Spigot-API-Patches/0217-Add-missing-extends.patch:91]: Mojang does not consider them flying.

File: Spigot-API-Patches/0217-Add-missing-extends.patch [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+diff --git a/src/main/java/org/bukkit/entity/Wither.java b/src/main/java/org/bukkit/entity/Wither.java
+index 426d3693317cd303d35d8203026b528d87e401d5..84f92c79a74ce335a4a192b10619238baa01a26d 100644
+--- a/src/main/java/org/bukkit/entity/Wither.java
++++ b/src/main/java/org/bukkit/entity/Wither.java
+@@ -5,5 +5,5 @@ import com.destroystokyo.paper.entity.RangedEntity;
+ /**
+  * Represents a Wither boss
+  */
+-public interface Wither extends Monster, Boss, RangedEntity { // Paper
++public interface Wither extends Monster, Boss, RangedEntity, Flying { // Paper - Withers fly
```
Comment (aikar) [inline Spigot-API-Patches/0217-Add-missing-extends.patch:104]: Not flying to Mojang.

**Review by aikar — CHANGES_REQUESTED** [review]
I'm a bit torn here. Ideally, Flying would not be bound to monster, but mojang does currently have 2 degrees of flying.

I feel like there should be a FlyingMonster extends Flying, Monster, and FlyingAnimal extends Flying, Animal

but currently bukkit is matching Flying -> FlyingMonster.

I question can we safely change Flying usages to FlyingMonster. 

However, the scary part is it really needs to align to mojangs definitions.

Currently, the flying interface is only used by animals, so thats just flying animals.

The Entityflying doesn't bind to anything monster oriented, but animals don't currently use it.

This one im iffy on doing too much ourselves, this may be best to PR to spigot with a breakout of FlyingMonster/FlyingAnimal type stuff.

Please join us on discord to talk about this more.

**Comment by ann4belle** [comment]
@aikar Your hesitance on doing this downstream from Spigot is understandable - I probably should've started with them, was just a bit confused on how to get access to do so. Found out after submitting this PR.

Once I'm more awake, I'll see if the Spigot devs would consider pushing a patch like this - if not, I'll come back and update this one w/ requested changes.

**Comment by Proximyst** [comment]
Source repository is deleted; unmergeable PR.

**Pull request closed without merging**